### ISSUE TITLE
Add profile resolution to global configuration (kdeglobals)

### ISSUE THREAD

**Issue by FelixResch** [ElektraInitiative/kconfig#15]
Should the profile for the kde globals be the same profile as for the other configuration? Or should we use a different one, maybe with a fallback option (if no globals profile is set, check if another profile is set, if so use this, else use the default profile)

@markus2330 what are your thoughts on this?

**Comment by markus2330** [comment]
What exactly do you mean with "profile"? Ideally we would treat everything with a simple `ksLookup` and do not handle any keys differently.

**Comment by FelixResch** [comment]
The keys are currently in the form `/sw/org/kde/<app_name>/#<major_version>/<profile>/`. For the kde globals I have used kdeglobals as `app_name` but currently always use "current" as `profile`.

The profile can be either set programmatically, by parameter `--elektra-profile` or by environment variable `KCONFIG_APP_PROFILE`. (I still need to document those, will do so tomorrow)

I am not using `ksLookup` because KConfig backends populate the key map on load and therefore I iterate over all keys.

As kde globals have a special meaning in KDE I wanted to ask, if we should make it possible to use different profiles for the applications and the global settings.

**Comment by markus2330** [comment]
> The profile can be either set programmatically, by parameter --elektra-profile or by environment variable KCONFIG_APP_PROFILE. (I still need to document those, will do so tomorrow)

Good, so this is something similar to what the profile plugin does: https://www.libelektra.org/plugins/profile

Please already start documenting this, so that we know which features are intended and how they work.

> I am not using ksLookup because KConfig backends populate the key map on load and therefore I iterate over all keys.

You can and should also call ksLookup even if you iterate over all keys. 

I just see that this is not explained in doc/tutorials/application-integration.md but it already says: "To make this work within your application make sure to always call `ksLookup` before using a value from Elektra."

> As kde globals have a special meaning in KDE I wanted to ask, if we should make it possible to use different profiles for the applications and the global settings.

Yes, I agree that it makes sense. Just because someone switches the profile of an application does not mean someone wants to switch to different kde globals.

**Comment by FelixResch** [comment]
> Please already start documenting this, so that we know which features are intended and how they work.

I added the documentation, should I create a separate PR for this branch?

> You can and should also call ksLookup even if you iterate over all keys.

I am now using ksLookup on the keys.

**Comment by markus2330** [comment]
> I added the documentation, should I create a separate PR for this branch?

Yes, please separately. We should merge #13 or #15 as fast as possible, as it is now quite chaotic what is reviewed and what is from you.

> I am now using ksLookup on the keys.

Thank you! Did you use cascading lookup (without namespace)? Did this change anything in the performance?

**Comment by FelixResch** [comment]
> Thank you! Did you use cascading lookup (without namespace)? Did this change anything in the performance?

Yes, I am using cascading namespaces.As far as I can tell, there is no noticeable change in performance.
